diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt

```diff
@@ -82,6 +96,32 @@ class HeaderCommentRule : Rule("header-comment") {
         }
     }
 
+    /**
+     * If corresponding rule is enabled, checks if header KDoc is positioned correctly and moves it in fix mode.
+     * @return true if position check is not needed or if header KDoc is positioned correctly or it was moved by fix mode
+     */
+    private fun checkHeaderKdocPosition(node: ASTNode): Boolean {
+        // AST always has PACKAGE_DIRECTIVE and IMPORT_LIST nodes, even if they are empty
+        if (configRules.isRuleEnabled(HEADER_NOT_BEFORE_PACKAGE) && node.findChildBefore(PACKAGE_DIRECTIVE, KDOC) == null) {
+            val firstKdoc = node.findChildAfter(IMPORT_LIST, KDOC)
+            if (firstKdoc != null && firstKdoc.treeParent.elementType == FILE) {
+                HEADER_NOT_BEFORE_PACKAGE.warnAndFix(configRules, emitWarn, isFixMode, fileName, firstKdoc.startOffset) {
+                    node.moveChildBefore(firstKdoc, node.getFirstChildWithType(PACKAGE_DIRECTIVE), true)
+                    // ensure there is no empty line between copyright and header kdoc
```

Comment:
static analysis will tell you that you have too many levels of nesting

Reply:
I see your point and will fix, but the check has been successful though...